File: src/test/programmers/java/level1/AthletesWhoDidNotFinish.java
package test.programmers.java.level1;

import java.util.HashMap;

public class AthletesWhoDidNotFinish {
    public static void main(String[] args) {
        String[] str = {"mislav", "stanko", "mislav", "ana"};
        String[] str2 = {"stanko", "ana", "mislav"};

        System.out.println(solution(str, str2));
    }

    static String solution(String[] participant, String[] completion) {
        String answer = "";

        HashMap<String, Integer> map = new HashMap<>();

        for(String player : participant) {
            map.put(player, map.getOrDefault(player, 0) + 1);
            // 키 값 밸류가 없으면 default + 1 있으면 원래 값 + 1
        }

        for(String player : completion) {
            map.put(player, map.get(player) - 1);
        }

        for(String key : map.keySet()) {
            if(map.get(key) != 0) {
                answer += key;
            }
        }

        return answer;
    }

}

File: src/test/programmers/python/level1/AthletesWhoDidNotFinish.py
def solution(participant, completion):
    dict_participant = {}
    for p in participant:
        dict_participant[p] = dict_participant.get(p, 0) + 1
    for c in completion:
        dict_participant[c] -= 1

    return "".join(i for i in dict_participant if dict_participant[i] != 0)

print(solution(["mislav", "stanko", "mislav", "ana"], ["stanko", "ana", "mislav"]))

# import collections
#
# def solution(participant, completion):
#     answer = collections.Counter(participant) - collections.Counter(completion)
#     return " ".join(list(answer.keys()))
#
# print(solution(["mislav", "stanko", "mislav", "ana"], ["stanko", "ana", "mislav"]))